### ISSUE TITLE
why the $state doesn't work in [$stateProvider.state > resolve]

### ISSUE THREAD

**Issue by axetroy** [angular-ui/ui-router#2572]
i use the params to control that when should render the view
so,i write some like this
```js
$stateProvider.state('admin',{
   ...
  resolve:{
    isAdmin:function($q,$state){
      ...
      if(...){
        $state.go('home');
        return $q.reject();
      }else{
        return $q.resolve();
      }
    }
  }
});
```

i want do something like this:
```js
if(false){
  //  if false don't render the view and go home page
 $state.go('home');
 return $q.reject();
}else{
  //  if true ,render the view
  return $q.resolve();
}
```

but i find it doesn't work at all;
this is a bug or limited by angular?
how can i fixed it?
ty~!

**Comment by joshrickert** [comment]
This is addressed in ui-router 1.0, but there are a few workarounds currently. The most "correct" way involves putting your `$state.go()` call in a `$stateChangeError` handler, but some people have also reported success just wrapping their `$state.go()` in a `$timeout`:

```
$timeout(function() {
        $state.go('home');
        return $q.reject();
});
```

**Comment by eddiemonge** [comment]
Try adding that code instead to an onEnter block.

Closing as this doesn't seem like a bug but instead an implementation issue.
